internal/provider/protocols_cifs_service_resource.go

```diff
@@ -0,0 +1,958 @@
+	var data *CifsServiceResourceModel
+
+	// Read Terraform plan data into the model
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &data)...)
+
+	var body interfaces.CifsServiceResourceBodyDataModelONTAP
+	errorHandler := utils.NewErrorHandler(ctx, &resp.Diagnostics)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	body.Name = data.Name.ValueString()
+	body.SVM.Name = data.SVMName.ValueString()
+
+	client, err := getRestClient(errorHandler, r.config, data.CxProfileName)
+	if err != nil {
+		// error reporting done inside NewClient
+		return
+	}
+	cluster, err := interfaces.GetCluster(errorHandler, *client)
+	if err != nil {
+		// error reporting done inside GetCluster
+		return
+	}
+	if cluster == nil {
+		errorHandler.MakeAndReportError("No cluster found", fmt.Sprintf("Cluster %s not found.", data.CxProfileName.ValueString()))
+		return
+	}
+	clusterVersion := strconv.Itoa(cluster.Version.Generation) + "." + strconv.Itoa(cluster.Version.Major)
+	var errors []string
+	// Create the resource
+	body.AdDomain.Fqdn = data.AdDomain.Fqdn.ValueString()
+	body.AdDomain.User = data.AdDomain.User.ValueString()
+	body.AdDomain.Password = data.AdDomain.Password.ValueString()
+	// optional fields
+	if !data.AdDomain.OrganizationalUnit.IsNull() {
+		body.AdDomain.OrganizationalUnit = data.AdDomain.OrganizationalUnit.ValueString()
+	}
+
+	// default value
+	body.Enabled = data.Enabled.ValueBool()
+
+	if !data.DefaultUnixUser.IsNull() {
+		body.DefaultUnixUser = data.DefaultUnixUser.ValueString()
+	}
+	if !data.Comment.IsNull() {
+		body.Comment = data.Comment.ValueString()
+	}
+
+	if !data.Netbios.IsUnknown() {
+		var netbios CifsNetbiosResourceModel
+		diags := data.Netbios.As(ctx, &netbios, basetypes.ObjectAsOptions{})
+		if diags.HasError() {
+			resp.Diagnostics.Append(diags...)
+			return
+		}
+		body.Netbios.Enabled = netbios.Enabled.ValueBool()
+		if !netbios.Aliases.IsNull() {
+			for _, e := range netbios.Aliases.Elements() {
+				body.Netbios.Aliases = append(body.Netbios.Aliases, e.(basetypes.StringValue).ValueString())
+			}
+		}
+		if !netbios.WinsServers.IsNull() {
+			windowServers := netbios.WinsServers.Elements()
+			body.Netbios.WinsServers = make([]string, len(windowServers))
+			for i, e := range windowServers {
+				body.Netbios.WinsServers[i] = e.String()
+			}
+		}
+	}
+
+	if !data.Security.IsUnknown() {
+		var security CifsSecurityResourceModel
+		diags := data.Security.As(ctx, &security, basetypes.ObjectAsOptions{})
+		if diags.HasError() {
+			resp.Diagnostics.Append(diags...)
+			return
+		}
+		body.Security.RestrictAnonymous = security.RestrictAnonymous.ValueString()
+		body.Security.SmbSigning = security.SmbSigning.ValueBool()
+		body.Security.SmbEncryption = security.SmbEncryption.ValueBool()
+		// kdc_encryption is only supported in 9.12 and earlier
+		if !security.KdcEncryption.IsNull() {
+			if CompareVersions(clusterVersion, "9.12") <= 0 {
```

Comment:
We don't really need a custom function for this

If you want higher than 9.8
if cluster.Version.Generation == 9 && cluster.Version.Major > 8 {

if you want 9.8 or higher 
if cluster.Version.Generation == 9 && cluster.Version.Major >= 8 {

if you want less than 9.8
if you want 9.8 or higher 
if cluster.Version.Generation == 9 && cluster.Version.Major < 8 {


Taking the version number, converting to a custom string, passing that off to a custom function that returns -1, 0, 1
complicates it more than i should be.